cmd/snap/cmd_export_key.go

```diff
@@ -73,7 +111,20 @@ func (x *cmdExportKey) Execute(args []string) error {
 			"name":                keyName,
 			"public-key-sha3-384": pubKey.ID(),
 			"since":               time.Now().UTC().Format(time.RFC3339),
-			// XXX: To support revocation, we need to check for matching known assertions and set a suitable revision if we find one.
+		}
+		if x.Revoke {
+			headers["until"] = headers["since"]
+			var authContext auth.AuthContext
+			sto := storeNew(nil, authContext)
+			// XXX A missing assertion isn't an error here,
+			// since we will just export a new one.  However, it
+			// isn't currently possible to distinguish between
+			// "assertion not found" and other errors at this
+			// point.
```

Comment:
confused by this comment, I think it should be distiguishable, I'm quite sure we have  a couple of places that do the distinction, but see the point about not using a Fetcher